### PR TITLE
Set Lazy Cookie

### PR DESCRIPTION
**Pull request by tergol1**
Set `VtexWorksapce` cookie.

### CODE REVIEW

File: react/core/main.tsx [hunk 1]
```diff
@@ -155,6 +155,12 @@ function validateRootComponent(rootName: string, extensions: Extensions) {
   }
 }
 
+function setLazyCookie(setCookie: string) {
+  if (setCookie !== null) {
+    document.cookie = setCookie
```
Comment (victorges) [inline react/core/main.tsx]: Is this really setting the raw value of the cookies? I thought the browser would have some API for accessing the cookies already parsed only.

Also, here you are *setting* the whole cookies, while I believe we should only *append* the lazy cookies to the existing ones. In that case, it'd probably be necessary to check if the cookie we are setting is already set.

File: react/core/main.tsx [hunk 2]
```diff
@@ -155,6 +155,12 @@ function validateRootComponent(rootName: string, extensions: Extensions) {
   }
 }
 
+function setLazyCookie(setCookie: string) {
+  if (setCookie !== null) {
```
Comment (victorges) [inline react/core/main.tsx]: You can check simply `if (setCookie)` which will also be false for `undefined`

File: react/core/main.tsx (hunk 1 above)
Comment (tergol1) [reply react/core/main.tsx]: Just for the last part, I think that this is not necessary as router won't set the header `X-Vtex-Lazy-Cookie` if it checks the existence of the cookie - we do the "cookie-stuff" inside `FindWorkspaceFromCookies`, where we first check if we can get the Workspace from cookies and then immediately returns the value.

Comment (tergol1) [reply react/core/main.tsx]: I don't know if I understood everything bothering you but I think that is everything ok. Just take a look here 
https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie

Reading it I concluded that the cookie is being set correctly and we don't have to actually care about the other ones that were set before (even after).

Comment (firstdoit) [reply react/core/main.tsx]: Yep. @victorges go back to your golang cave 😘 hahahahaha 😬 😂

**Pull request merged**